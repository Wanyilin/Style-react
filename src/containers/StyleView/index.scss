@import 'src/theme';

.style-view {
	width: 216px;
	background: #101011;
	padding: $MainPadding;

	.style-list {
		@include BorderButtomLine(solid);
		padding-bottom: $BoxPadding;
		margin-bottom: $BoxMargin;
	}
	.style-list:last-child {
		border-bottom: 0;
		padding-bottom: 0;
		margin-bottom: 0;
	}
	.category-title {
		@include FontStyle($CategoryTitleSize, $CategoryTitleWeight, $MainFontColor);
		letter-spacing: 0.01em;
		margin-bottom: $ListItemMargin;
	}
	.style-list-group {
		display: flex;
		flex-direction: column;
	}

	.style-list-group:last-child {
		margin-bottom: 0;

		&::after {
			content: none;
		}
	}

	.style-list-group::after {
		align-self: center;
    width: 200px;
    margin: $BoxMargin 0;
    display: block;
    content: '';
    border-bottom: 1px #45454F dashed;
		@include BorderButtomLine(dashed)
	}
}