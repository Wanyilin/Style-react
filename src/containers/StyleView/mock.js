const stylesList = {
	font: {
		fontFamily: 'Inter',
		fontStyle: 'Medium',
		fontSize: 24,
		fontSizeUnit: 'px',
		horizontalAlignment: 'LEFT',
		verticalAlignment: 'TOP',
		lineHeight: 1,
		paragraphSpacing: 0.2,
		characterSpacing: 0.01,
		wordSpacing: 0.01,
		paragraphIndent: 0.1,
		textDecoration: 'UNDERLINE',
		textCases: 'UPPERCASE',
		textDirection: 'RTL'
	},
	fills: [
		{
			paintType: 'SOLID',
			color: [1, 0.86, 0.15, 0.2],
			opacity: 0.2,
			blendMode: 'PASS_THROUGH'
		},
		{
			paintType: 'GRADIENT',
			gradientType: 'LINEAR',
			gradientStops: [
					{ color: [0.9961, 0.5059, 0.5059, 1], position: 0 },
					{ color: [1, 0.6941, 0.4118, 1], position: 0.3 },
					{ color: [1, 0.9412, 0.4118, 0.7], position: 1 }
			],
			opacity: 1,
			blendMode: 'LIGHTBURN'
		},
		{
			paintType: 'IMAGE',
			blendMode: 'PASS_THROUGH',
			fillType: 'FIT',
			x: 9,
			y: 0,
			width: 240,
			height: 890,
			rotation: 0.08726646259971647,
			opacity: 1
		}
	],
	strokes: [
			{
				paintType: 'SOLID',
				color: [0.149, 0.6941, 1, 1],
				opacity: 1,
				blendMode: 'OVERLAY',
				width: 1,
				offset: -0.5
			},
			{
				paintType: 'GRADIENT',
				gradientType: 'LINEAR',
				gradientStops: [
					{ color: [0.9961, 0.5059, 0.5059, 1], position: 0 },
					{ color: [1, 0.6941, 0.4118, 1], position: 0.3 },
					{ color: [1, 0.9412, 0.4118, 0.7], position: 1 }
				],
				opacity: 1,
				blendMode: 'NORMAL',
				width: 2,
				offset: 0
			}
	]
};

export { stylesList };