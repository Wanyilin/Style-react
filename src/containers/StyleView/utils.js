import {
	GRADIENT_OBJ,
	GRADIENT_TYPE,
	STYLE_VALUE_TYPE,
	FONT_OBJ,
	EXCLUD_LIST,
} from 'src/utils/constants';


const mapStyleItems = (list) => {
	const mapStyles = Object.keys(list).map(styleName => {
		switch (list[styleName]) {
			case STYLE_VALUE_TYPE.GRADIENT:
				return ({
					name: GRADIENT_TYPE[list[GRADIENT_OBJ.GradientType]],
					value: list[GRADIENT_OBJ.GradientSpots],
					type: STYLE_VALUE_TYPE.GRADIENT,
				})
			case STYLE_VALUE_TYPE.IMAGE:
				return ({
					name: list[styleName],
					value: list?.url,
					type: STYLE_VALUE_TYPE.IMAGE,
				})
			case STYLE_VALUE_TYPE.SOLID:
				return ({
					name: '',
					value: list.color,
					type: STYLE_VALUE_TYPE.SOLID,
				})
			case FONT_OBJ.FONT_SIZE:
				return ({
					name: styleName,
					value: list[styleName],
					unit: list[FONT_OBJ.FONT_UNIT],
					tyep: STYLE_VALUE_TYPE.STRING,
				})
			default:
				return ({
					name: styleName,
					value: list[styleName],
					type: STYLE_VALUE_TYPE.STRING,
				})
		}
	});
	const filterItems = mapStyles.filter(({ name }) => (
		!EXCLUD_LIST.includes(name)
	))
	return filterItems;
};

export { mapStyleItems };
