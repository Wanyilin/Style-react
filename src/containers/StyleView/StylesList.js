import React from 'react';
import ListItem from 'src/components/ListItem';
import ColorGradient from 'src/components/ColorGradient';
import PaintPreview from 'src/components/PaintPreview';
import { STYLE_VALUE_TYPE } from 'src/utils/constants';
import { mapStyleItems } from './utils';

const ItemComponent = ({ type, value, name }) => {
	switch (type) {
		case STYLE_VALUE_TYPE.GRADIENT:
			return <ColorGradient key={`${name}-${value}`} type={type} value={value} name={name} />;
		case STYLE_VALUE_TYPE.IMAGE:
		case STYLE_VALUE_TYPE.SOLID:
			return <PaintPreview key={`${name}-${value}`} type={type} value={value} name={name} />;
		default:
			return <ListItem key={`${name}-${value}`} type={type} value={value} name={name} />
	}
}

const StylesList = ({
	stylesList,
	className,
}) => {
	const mappedItems = mapStyleItems(stylesList);
	return (
		<div className={className}>
			{mappedItems.map(ItemComponent)}
		</div>
	)
};

export default StylesList;
