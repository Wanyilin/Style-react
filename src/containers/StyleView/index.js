import React from 'react';
import { convertCamel } from 'src/utils/formatter';
import StylesList from './StylesList';
import { stylesList } from './mock';
import './index.scss';

const stylesListGroup = groups => (
	groups.map(stylesList => (
		<StylesList
			className="style-list-group"
			key={stylesList}
			stylesList={stylesList}
		/>
	))
);

const StyleView = () => {
	const categoriesList = Object.keys(stylesList);
	return (
		<div  className="style-view">
			{
				categoriesList.map(cat => (
					<div key={cat} className="style-list">
						<div className="category-title">{convertCamel(cat)}</div>
						{Array.isArray(stylesList[cat]) ? (
							stylesListGroup(stylesList[cat])
						) : (
							<StylesList stylesList={stylesList[cat]}/>
						)}
					</div>
				))
			}
		</div>
	)
}

export default StyleView;
