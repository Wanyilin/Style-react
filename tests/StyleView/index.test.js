import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import StyleView from '../../src/containers/StyleView';

jest.mock('../../src/containers/StyleView/StylesList', () => () => (
	'Styles List'
));

describe('StyleView', () => {
	test('Should match snapshot', () => {
		const component = renderer.create(<StyleView />);
		let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
	});
	test('Should render Style view correctly', () => {
		const { container, getByText } = render(
			<StyleView />,
		);
		const categoriesElems = container.getElementsByClassName('category-title');
		expect(categoriesElems.length).toEqual(3);
		expect(getByText('Styles List')).toBeInTheDocument();
	})
});
