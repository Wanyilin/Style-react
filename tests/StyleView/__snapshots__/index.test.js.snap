// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleView Should match snapshop 1`] = `
<div
  className="style-view"
>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Font
    </div>
    Styles List
  </div>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Fills
    </div>
    Styles List
    Styles List
    Styles List
  </div>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Strokes
    </div>
    Styles List
    Styles List
  </div>
</div>
`;

exports[`StyleView Should match snapshot 1`] = `
<div
  className="style-view"
>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Font
    </div>
    Styles List
  </div>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Fills
    </div>
    Styles List
    Styles List
    Styles List
  </div>
  <div
    className="style-list"
  >
    <div
      className="category-title"
    >
      Strokes
    </div>
    Styles List
    Styles List
  </div>
</div>
`;
